diff -Nur st.orig/config.h st/config.h
--- st.orig/config.h	2017-02-06 10:31:26.888168676 +0100
+++ st/config.h	2017-02-06 10:35:34.134143305 +0100
@@ -5,7 +5,7 @@
  *
  * font: see http://freedesktop.org/software/fontconfig/fontconfig-user.html
  */
-static char font[] = "Liberation Mono:pixelsize=12:antialias=true:autohint=true";
+static char font[] = "Terminus:pixelsize=10:antialias=false:autohint=true";
 static int borderpx = 2;
 
 /*
@@ -84,29 +84,29 @@
 
 /* Terminal colors (16 first used in escape sequence) */
 static const char *colorname[] = {
-	/* 8 normal colors */
-	"black",
-	"red3",
-	"green3",
-	"yellow3",
-	"blue2",
-	"magenta3",
-	"cyan3",
-	"gray90",
-
-	/* 8 bright colors */
-	"gray50",
-	"red",
-	"green",
-	"yellow",
-	"#5c5cff",
-	"magenta",
-	"cyan",
-	"white",
-
-	[255] = 0,
-
-	/* more colors can be added after 255 to use with DefaultXX */
-	"#cccccc",
-	"#555555",
+        /* 8 normal colors */
+  "#000000",
+  "#cc0000",
+  "#4e9a06",
+  "#c4a000",
+  "#3465a4",
+  "#75507b",
+  "#06989a",
+  "#d3d7cf",
+
+  /* 8 bright colors */
+  "#555753",
+  "#ef2929",
+  "#8ae234",
+  "#fce94f",
+  "#739fcf",
+  "#ad7fa8",
+  "#34e2e2",
+  "#eeeeec",
+
+        [255] = 0,
+
+        /* more colors can be added after 255 to use with DefaultXX */
+        "#cccccc",
+        "#555555",
 };
