From 66c35797a515d14a6482fbb9da2f58b407083591 Mon Sep 17 00:00:00 2001
From: Gwenhael Le Moine <gwenhael.le.moine@gmail.com>
Date: Mon, 2 Sep 2024 13:35:20 +0200
Subject: [PATCH] fix compilation with kernel 6.11.0-rc6

---
 nct6687.c | 7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

diff --git a/nct6687.c b/nct6687.c
index c23889f..de1fa4c 100644
--- a/nct6687.c
+++ b/nct6687.c
@@ -37,9 +37,6 @@
 #include <linux/platform_device.h>
 #include <linux/slab.h>
 
-#define MIN(a,b) (((a)<(b))?(a):(b))
-#define MAX(a,b) (((a)>(b))?(a):(b))
-
 enum kinds
 {
 	nct6683,
@@ -1034,7 +1031,7 @@ static void nct6687_setup_pwm(struct nct6687_data *data)
 	}
 }
 
-static int nct6687_remove(struct platform_device *pdev)
+static void nct6687_remove(struct platform_device *pdev)
 {
 	struct device *dev = &pdev->dev;
 	struct nct6687_data *data = dev_get_drvdata(dev);
@@ -1048,8 +1045,6 @@ static int nct6687_remove(struct platform_device *pdev)
 	}
 
 	mutex_unlock(&data->update_lock);
-
-	return 0;
 }
 
 static int nct6687_probe(struct platform_device *pdev)
