--- /etc/X11/xdm/./Xsetup_0	2009-05-30 08:47:39.000000000 +0700
+++ ./Xsetup_0	2009-11-19 15:34:23.000000000 +0700
@@ -8,11 +8,50 @@
 	/usr/bin/xmodmap $sysmodmap
 fi
 
-# If the nifty KDE background loader is around, use it:
-if [ -x /opt/kde/bin/kdmdesktop ]; then
-	/opt/kde/bin/kdmdesktop
-elif [ -x /usr/bin/kdmdesktop ]; then
-	/usr/bin/kdmdesktop
-else
-	/usr/bin/xsetroot -solid SteelBlue
-fi
+PATH="/bin:/sbin:/usr/bin:/usr/sbin:/usr/X11R6/bin:/usr/local/bin";
+
+wish8.4 -name Powerbox <<EOF &
+wm geometry . -0-0
+button .halt                 \
+-bg black                    \
+-activebackground black      \
+-activeforeground #e0e0e0    \
+-fg #4f4f5f                  \
+-highlightbackground black   \
+-font aqui\ 10               \
+-text halt                   \
+-borderwidth 0               \
+-cursor left_ptr             \
+-command { exec halt 2>/dev/console >/dev/console }
+button .reboot               \
+-bg black                    \
+-activebackground black      \
+-activeforeground #e0e0e0    \
+-fg #4f4f5f                  \
+-highlightbackground black   \
+-font aqui\ 10               \
+-text reboot                 \
+-borderwidth 0               \
+-cursor left_ptr             \
+-command { exec reboot 2>/dev/console >/dev/console }
+button .console              \
+-bg black                    \
+-activebackground black      \
+-activeforeground #e0e0e0    \
+-fg #4f4f5f                  \
+-highlightbackground black   \
+-font aqui\ 10               \
+-text console                \
+-borderwidth 0               \
+-cursor left_ptr             \
+-command { exec killall xdm 2>/dev/console >/dev/console }
+pack .console .halt .reboot -side left -fill both
+EOF
+
+umask 022
+# saving PID of Powerbox process
+echo $! > /var/run/Xsession.pid
+
+#xv -root -quit -viewonly /usr/X11R6/lib/X11/xdm/pixmaps/FreeBSD.jpg
+fbsetbg -f /usr/X11R6/lib/X11/xdm/pixmaps/headphones.jpg
+#xsnow &
